api-server/src/server/boot/challenge.js

```diff
@@ -273,63 +272,60 @@ export function isValidChallengeCompletion(req, res, next) {
   return next();
 }
 
-export function modernChallengeCompleted(req, res, next) {
+export async function modernChallengeCompleted(req, res, next) {
   const user = req.user;
-  return user
-    .getCompletedChallenges$()
-    .flatMap(() => {
-      const completedDate = Date.now();
-      const { id, files, challengeType } = req.body;
-
-      const completedChallenge = {
-        id,
-        files,
-        completedDate
-      };
-
-      // if multifile cert project
-      if (challengeType === 14) {
-        completedChallenge.isManuallyApproved = false;
-        user.needsModeration = true;
-      }
 
-      // We only need to know the challenge type if it's a project. If it's a
-      // step or normal challenge we can avoid storing in the database.
-      if (
-        jsCertProjectIds.includes(id) ||
-        multifileCertProjectIds.includes(id)
-      ) {
-        completedChallenge.challengeType = challengeType;
-      }
+  try {
+    // This is an ugly way to update `user.completedChallenges`
+    await user.getCompletedChallenges$().toPromise();
+  } catch (e) {
+    return next(e);
+  }
 
-      const { alreadyCompleted, savedChallenges, updateData } = buildUserUpdate(
-        user,
-        id,
-        completedChallenge
-      );
+  const completedDate = Date.now();
+  const { id, files, challengeType } = req.body;
 
-      const points = alreadyCompleted ? user.points : user.points + 1;
-      const updatePromise = new Promise((resolve, reject) =>
-        user.updateAttributes(updateData, err => {
-          if (err) {
-            return reject(err);
-          }
-          return resolve();
-        })
-      );
-      return Observable.fromPromise(updatePromise).map(() => {
-        return res.json({
-          points,
-          alreadyCompleted,
-          completedDate,
-          savedChallenges
-        });
-      });
-    })
-    .subscribe(() => {}, next);
+  const completedChallenge = {
+    id,
+    files,
+    completedDate
+  };
+
+  // if multifile cert project
+  if (challengeType === 14) {
+    completedChallenge.isManuallyApproved = false;
+    user.needsModeration = true;
+  }
+
+  // We only need to know the challenge type if it's a project. If it's a
+  // step or normal challenge we can avoid storing in the database.
+  if (jsCertProjectIds.includes(id) || multifileCertProjectIds.includes(id)) {
+    completedChallenge.challengeType = challengeType;
+  }
+
+  const { alreadyCompleted, savedChallenges, updateData } = buildUserUpdate(
+    user,
+    id,
+    completedChallenge
+  );
+
+  const points = alreadyCompleted ? user.points : user.points + 1;
+
+  user.updateAttributes(updateData, (err, _updatedUser) => {
```

Comment:
```suggestion
  user.updateAttributes(updateData, err => {
```
Seems redundant if we're not doing anything with the updated user.

Reply:
True, but I just like knowing it is there. Still want me to remove it?

Reply:
I don't mind it that much.